### DISCUSSION TITLE
Webhook parallel requests processing

### DISCUSSION THREAD

**Discussion by paradise** [telegraf/telegraf#1859] [General]
Hello! I'm facing an issue with Telegram's webhook requests. Currently, the requests are being sent one by one instead of in parallel, even when multiple people are using my bot.

My message processing time is approximately 2-3 seconds. So, when person A sends a message to my bot, the bot starts processing that message. However, if person B sends a message to my bot at the same time (in a different private chat), I don't receive the webhook from Telegram until I finish processing person A's request.

I'm wondering if there's something I'm doing wrong. Is there a way to configure Telegraf to allow Telegram to send multiple webhook requests from different people simultaneously?

**Comment by MKRhere** [comment]
There's no way to configure this. Bot API will always send us updates per chat in series. Telegraf then waits for middleware stack to finish processing to respond with success.

To make Telegraf not wait for completion, you can use this hack before your longrunning handlers:

```TS
bot.use((ctx, next) => {
  next().catch(errorHandler);
});
```

Since we don't await or return next, Telegraf will not wait for subsequent handlers before responding with a success, prompting Telegram to send the next update. Telegraf cannot catch errors in them either since they're not awaited, which is why we manually attach the error handler here.

A better way would be to have a queue where you push updates and handle them with a worker.
